independent-projects/bootstrap/maven-resolver/src/main/java/io/quarkus/bootstrap/resolver/maven/DeploymentInjectingDependencyVisitor.java

```diff
@@ -44,11 +59,16 @@ public static Artifact getRuntimeArtifact(DependencyNode dep) {
     private final MavenArtifactResolver resolver;
     private final List<Dependency> managedDeps;
     private final List<RemoteRepository> mainRepos;
+    private final AppModel.Builder appBuilder;
 
-    private boolean collectExtensions = true;
+    private boolean collectingTopRuntimeNodes = true;
+    private List<ExtensionDependency> topExtensionDeps = new ArrayList<>();
+    private ExtensionDependency lastVisitedRuntimeExtNode;
+    private Map<AppArtifactKey, ExtensionInfo> allExtensions = new HashMap<>();
+    private List<ConditionalDependency> conditionalDepsToProcess = new ArrayList<>();
+    private List<Collection<Exclusion>> exclusionStack = new ArrayList<>();
```

Comment:
If you're always pushing and popping, how about:

```java
private final Deque<Collection<Exclusion>> exclusionStack = new ArrayDeque<>();
```

Also some of these fields should probably be `final` I think?

Reply:
Done, thanks!